pp_ctl.c

```diff
@@ -4593,6 +4614,22 @@ PP(pp_leaveeval)
     return retop;
 }
 
+PP(pp_catch)
+{
+    dTARGET;
+
+    if(!SvROK(ERRSV) && !SvTRUE(ERRSV)) {
```

Comment:
This won't catch `die 0;`. I think it should use the same logic as [die](https://github.com/perl/perl5/blob/v5.33.6/pp_sys.c#L475), i.e. `if (SvROK(exsv) || (SvPV_const(exsv, len), len))`.

Reply:
Turns out that `die 0` was fine, because of the appended "... at FILE line LINE." string. But I've added a unit test to ensure that remains the case.